test/Mono.Linker.Tests.Cases/Reflection/ConstructorUsedViaReflection.cs

```diff
@@ -10,226 +11,365 @@ public class ConstructorUsedViaReflection
 		}
 
-		[RecognizedReflectionAccessPattern (
-			typeof (Type), nameof (Type.GetConstructor), new Type[] { typeof (Type[]) },
-			typeof (IntegerParameterConstructor), ".ctor", new Type[0])]
 		[Kept]
-		static void TestWithIntegerParameter ()
+		private class GetConstructor_Types
 		{
-			var constructor = typeof (IntegerParameterConstructor).GetConstructor (new Type[] { typeof (int) });
-			constructor.Invoke (null, new object[] { });
-		}
+			[Kept]
+			public GetConstructor_Types ()
+			{ }
 
-		[RecognizedReflectionAccessPattern (
-			typeof (Type), nameof (Type.GetConstructor), new Type[] { typeof (BindingFlags), typeof (Binder), typeof (Type[]), typeof (ParameterModifier[]) },
-			typeof (OnlyUsedViaReflection), ".ctor", new Type[0])]
-		[Kept]
-		static void TestWithBindingFlags ()
-		{
-			var constructor = typeof (OnlyUsedViaReflection).GetConstructor (BindingFlags.Public, GetNullValue ("some argument", 2, 3), new Type[] { }, new ParameterModifier[] { });
-			constructor.Invoke (null, new object[] { });
-		}
+			[Kept]
+			public GetConstructor_Types (int i)
+			{ }
 
-		[RecognizedReflectionAccessPattern (typeof (Type), nameof (Type.GetConstructor), new Type[] { typeof (BindingFlags), typeof (Binder), typeof (Type[]), typeof (ParameterModifier[]) },
-			typeof (UnknownBindingFlags), ".ctor", new Type[0])]
-		[Kept]
-		static void TestWithUnknownBindingFlags (BindingFlags bindingFlags)
-		{
-			// Since the binding flags are not known linker should mark all constructors on the type
-			var constructor = typeof (UnknownBindingFlags).GetConstructor (bindingFlags, GetNullValue ("some argument", 2, 3), new Type[] { }, new ParameterModifier[] { });
-			constructor.Invoke (null, new object[] { });
-		}
+			private GetConstructor_Types (string foo)
+			{ }
 
-		[RecognizedReflectionAccessPattern (
-			typeof (Type), nameof (Type.GetConstructor), new Type[] { typeof (BindingFlags), typeof (Binder), typeof (CallingConventions), typeof (Type[]), typeof (ParameterModifier[]) },
-			typeof (CallingConventionConstructor), ".ctor", new Type[0])]
-		[Kept]
-		static void TestWithCallingConvention ()
-		{
-			var constructor = typeof (CallingConventionConstructor).GetConstructor (BindingFlags.Public, GetNullValue ("some argument", 2, 3), CallingConventions.HasThis, new Type[] { }, new ParameterModifier[] { });
-			constructor.Invoke (null, new object[] { });
-		}
+			protected GetConstructor_Types (string foo, string bar)
+			{ }
 
-		[Kept]
-		static void TestNullType ()
-		{
-			Type type = null;
-			var constructor = type.GetConstructor (new Type[] { });
-		}
+			[Kept]
+			class EmptyTypes
+			{
+				[Kept]
+				public EmptyTypes () { }
+				public EmptyTypes (int i) { }
+			}
 
-		[Kept]
-		static Type FindType ()
-		{
-			return null;
-		}
+			[Kept]
+			public static void Test ()
+			{
+				TestConstructorWithTypes_EmptyTypes ();
+				TestConstructorWithTypes_NonEmptyTypes ();
+				TestConstructorWithTypes_EmptyTypes_DataFlow (null);
```

Comment:
Any specific reason you used null instead of a real type for dataflow tests?

Reply:
It doesn't matter - the test doesn't care what's passed in here. But I changed it so that if one day we actually execute the tests they don't crash.